tests/dash/gnmi_utils.py

```diff
@@ -0,0 +1,377 @@
+                        -days 1825 \
+                        -subj '/CN=test.gnmi.sonic' \
+                        -out %s" % (env.work_dir+env.gnmi_ca_key, env.work_dir+env.gnmi_ca_cert)
+    localhost.shell(local_command)
+
+    # Create server key
+    local_command = "openssl genrsa -out %s 2048" % (env.work_dir+env.gnmi_server_key)
+    localhost.shell(local_command)
+
+    # Create server CSR
+    local_command = "openssl req \
+                        -new \
+                        -key %s \
+                        -subj '/CN=test.server.gnmi.sonic' \
+                        -out %s" % (
+                            env.work_dir+env.gnmi_server_key,
+                            env.work_dir+"gnmiserver.csr")
+    localhost.shell(local_command)
+
+    # Sign server certificate
+    create_ext_conf(duthost.mgmt_ip, env.work_dir+"extfile.cnf")
+    local_command = "openssl x509 \
+                        -req \
+                        -in %s \
+                        -CA %s \
+                        -CAkey %s \
+                        -CAcreateserial \
+                        -out %s \
+                        -days 825 \
+                        -sha256 \
+                        -extensions req_ext -extfile %s" % (
+                            env.work_dir+"gnmiserver.csr",
+                            env.work_dir+env.gnmi_ca_cert,
+                            env.work_dir+env.gnmi_ca_key,
+                            env.work_dir+env.gnmi_server_cert,
+                            env.work_dir+"extfile.cnf")
+    localhost.shell(local_command)
+
+    # Create client key
+    local_command = "openssl genrsa -out %s 2048" % (env.work_dir+env.gnmi_client_key)
+    localhost.shell(local_command)
+
+    # Create client CSR
+    local_command = "openssl req \
+                        -new \
+                        -key %s \
+                        -subj '/CN=test.client.gnmi.sonic' \
+                        -out %s" % (
+                            env.work_dir+env.gnmi_client_key,
+                            env.work_dir+"gnmiclient.csr")
+    localhost.shell(local_command)
+
+    # Sign client certificate
+    local_command = "openssl x509 \
+                        -req \
+                        -in %s \
+                        -CA %s \
+                        -CAkey %s \
+                        -CAcreateserial \
+                        -out %s \
+                        -days 825 \
+                        -sha256" % (
+                            env.work_dir+"gnmiclient.csr",
+                            env.work_dir+env.gnmi_ca_cert,
+                            env.work_dir+env.gnmi_ca_key,
+                            env.work_dir+env.gnmi_client_cert)
+    localhost.shell(local_command)
+
+
+def apply_gnmi_cert(duthost, ptfhost):
+    env = GNMIEnvironment(duthost)
+    # Copy CA certificate and server certificate over to the DUT
+    duthost.copy(src=env.work_dir+env.gnmi_ca_cert, dest='/etc/sonic/telemetry/')
```

Comment:
`/etc/sonic/telemetry/` path is used in 6 places. I suggest putting it into a "constant" variable

Reply:
updated